src/guide/security/generating-cryptographic-keys.md

```diff
@@ -0,0 +1,225 @@
+
+  - `bls_normal` — Boneh-Lynn-Shacham algorithm with a _standard_ parameter configuration. This configuration of the <abbr title ="Boneh-Lynn-Shacham">BLS</abbr> cryptographic scheme offers a balanced approach between efficiency and security, making it suitable for a wide range of applications in blockchain and cryptographic protocols.
+  > Learn more:\
+  [BLS digital signature (Wikipedia)](https://en.wikipedia.org/wiki/BLS_digital_signature)
+
+- `--seed` (`-s`): Specifies a string that serves as a deterministic starting point for the key pair generation. If a seed string is specified, `kagami` will generate the same key for the same string. If no seed is specified, a random `seed` value is chosen, and each invocation of `kagami crypto` will result in a different key. This parameter accepts a valid string of [Unicode](https://home.unicode.org/) characters. For example, the seed string can contain not only numeric and latin, but also cyrillic, logographic (e.g., Japanese kanji characters) and ideographic (e.g., emojis) characters, as well as any font-related variations of those characters introduced to Unicode over the years.
+
+  ::: tip Note
+
+  If one chooses to use a `seed` it must be treated as if it were a password: more randomness and longer seed strings make the cryptographic keys more resilient to [dictionary attacks](https://en.wikipedia.org/wiki/Dictionary_attack).
+
+  :::
+
+- `--private-key` (`-p`): Specifies an existing private key as a string in the [`multihash`](https://github.com/multiformats/multihash) format that is used as to generate a _public_ key.
+
+- `--json` (`-j`): Specifies that the output must be generated in the JSON format, which is mostly useful for copy-and-pasting into the `config.json`.
+
+- `--compact` (`-c`): Specifies that the output private and public keys are displayed on separate lines and are not labeled.
+
+### Examples
+
+::: code-group Examples
+
+```bash [No parameters]
+
+# Input
+$ cargo run --bin kagami crypto
+
+# Possible Output (same layout, different keys)
+Public key (multihash): "ed01206B0F56F58761060056355DBA0E0FC489CFB2F974481ED64873082E6032796235"
+Private key (ed25519): "F71EA9D897C4338CBF4F1DC7B492AAD0BF6CE896B803D7CDB9CF25ECC15109826B0F56F58761060056355DBA0E0FC489CFB2F974481ED64873082E6032796235"
+
+```
+
+```bash [--algorithm]
+
+# Input
+$ cargo run --bin kagami crypto -a secp256k1
+
+# Possible output
+Public key (multihash): "e70121022A9D6E0D54022C0E2752E43ADD91ADA28259E1F2CE0C6D4E9183FB2882DE6749"
+Private key (secp256k1): "7687B1433FB6731E6DC635A376B3EB3B5FCD1E02C9775C1642E7FD5DA035EC75"
+
+```
+
+```bash [--seed]
+
+# Input
+$ cargo run --bin kagami crypto -s 1729
+
+# Exact output
+Public key (multihash): "ed0120B678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+Private key (ed25519): "2669BB1099477B970E1D7D7C54E345A64A54213FCFBA2465CBCD6D4E5091A71DB678073CFAE6E247A58B442661C7DA0E13BAC5031CBC6343EF566B8718D47D04"
+
+```
+
+```bash [--private-key]
+
```

Comment:
Do not use extra spaces in code fences. Instead of

````md
```bash

$ kagami crypto

```
````

just write

````md
```bash
$ kagami crypto
```
````